java/ql/test/query-tests/security/CWE-297/UnsafeHostnameVerification.java

```diff
@@ -0,0 +1,99 @@
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLSession;
+import java.security.cert.Certificate;
+
+public class UnsafeHostnameVerification {
+
+	private static final boolean DISABLE_VERIFICATION = true;
+
+	/**
+	 * Test the implementation of trusting all hostnames as an anonymous class
+	 */
+	public void testTrustAllHostnameOfAnonymousClass() {
+		HttpsURLConnection.setDefaultHostnameVerifier(new HostnameVerifier() {
+			@Override
+			public boolean verify(String hostname, SSLSession session) {
+				return true; // BAD, always returns true
+			}
+		});
+	}
+
+	/**
+	 * Test the implementation of trusting all hostnames as a lambda.
+	 */
+	public void testTrustAllHostnameLambda() {
+		HttpsURLConnection.setDefaultHostnameVerifier((name, s) -> true); // BAD, always returns true
+	}
+
+	/**
+	 * Test an all-trusting hostname verifier that is guarded by a flag
+	 */
+	public void testGuardedByFlagTrustAllHostname() {
+		if (DISABLE_VERIFICATION) {
+			HttpsURLConnection.setDefaultHostnameVerifier(ALLOW_ALL_HOSTNAME_VERIFIER); // GOOD: The all-trusting
+																						// hostname verifier is guarded
+																						// by a feature flag
+		}
+	}
+
+	public void testGuardedByFlagAccrossCalls() {
+		if (DISABLE_VERIFICATION) {
+			functionThatActuallyDisablesVerification();
+		}
+	}
+
+	private void functionThatActuallyDisablesVerification() {
+		HttpsURLConnection.setDefaultHostnameVerifier((name, s) -> true); // GOOD [but detected as BAD], because we only
+																			// check guards inside a function
```

Comment:
There's some bad indentation in this file, e.g. here and a few lines above. Please clean up the whitespace issues (e.g. by removing tabs).

Reply:
File has been autoformatted using 4 spaces.